src/renderer/gl_engine/tvgGlRenderTask.cpp

```diff
@@ -106,16 +106,25 @@ GlComposeTask::~GlComposeTask()
     mTasks.clear();
 }
 
+void GlComposeTask::clear(GLfloat clearColor[4]) {
+    mClearColor[0] = clearColor[0];
+    mClearColor[1] = clearColor[1];
+    mClearColor[2] = clearColor[2];
+    mClearColor[3] = clearColor[3];
+    mClear = true;
+}
+
 void GlComposeTask::run()
 {
     GL_CHECK(glBindFramebuffer(GL_FRAMEBUFFER, getSelfFbo()));
 
     // clear this fbo
     GLenum color_buffer = GL_COLOR_ATTACHMENT0;
-    const float transparent[] = {0.f, 0.f, 0.f, 0.f};
-
     GL_CHECK(glDrawBuffers(1, &color_buffer));
-    GL_CHECK(glClearBufferfv(GL_COLOR, 0, transparent));
```

Comment:
GlComposeTask always need to clear the color-buffer since it may used by a off-screen rendering.
So I think, just change keep the default value for the mClearColor as transparent, and change it if outside user need a different background color

Reply:
Since the graphics library (GL) is under development, either way is fine with me. However, so far, raster engines don't need to determine this behavior like OpenGL does. We open the `clear()` API, and users can choose it according to their demand—simple. The `clear()` function is not designed to fill the background color; it is designed for composing with other existing(previous) images. If we clear it, they have no choice but to do so.

Reply:
Now, in the GL backend, the canvas's content is rendered to a off-screen framebuffer first, and then blit to the outside framebuffer. If need to composite with outside framebuffer ,the only change need to do is **render the texture** to outside framebuffer (like compose the SubRenderPass to the main RenderPass) not **blit the texture**.

Reply:
Ultimately, we will consider the target to be either the main or intermediate target. Secondly, even though the GL target serves as the framebuffer object (FBO), users might want to add more elements on top of the previous image. IMHO, There are no clear benefits to clearing the buffer by the engine itself, but just add more restrcited usage.

Reply:
An even more important point is that a consistent behavior through the raster backends for the clear() API is required for ThorVG.

Reply:
Maybe we can define canvas loading operations (clear, load) like WebGPU and let the backend implement the behavior